### PR TITLE
[ADD] DCOS-1323 with tests

### PR DESCRIPTION
**Pull request by truskovskiyk**
Rewrite task DCOS-1323 with remarks from https://github.com/mesosphere/dcos-cli/pull/240

### CODE REVIEW

**Comment by ytalashkocv** [comment]
@ssk2 Please review

File: cli/tests/integrations/test_help.py [hunk 1]
```diff
@@ -49,3 +50,263 @@ def test_list():
 
     assert_command(['dcos', 'help'],
                    stdout=stdout)
+
+
+def test_help_config():
+    stdout = """\
+Get and set DCOS CLI configuration properties
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: Can we put the usage text in a data file?

File: cli/tests/integrations/test_help.py [hunk 2]
```diff
@@ -49,3 +50,263 @@ def test_list():
...
+    <task-id>                   The task id
+
+""".encode('utf-8')
+    assert_command(['dcos', 'help', 'marathon'],
+                   stdout=stdout)
+
+
+def test_help_node():
+    stdout = """Manage DCOS nodes
+
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: Can we put the usage text in a data file please?

File: cli/tests/integrations/test_help.py [hunk 3]
```diff
@@ -49,3 +50,263 @@ def test_list():
...
+    --version     Show version
+
+""".encode('utf-8')
+    assert_command(['dcos', 'help', 'node'],
+                   stdout=stdout)
+
+
+def test_help_package():
+    stdout = """Install and manage DCOS software packages
+
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: Can we put the usage text in a data file please?

File: cli/tests/integrations/test_help.py [hunk 4]
```diff
@@ -49,3 +50,263 @@ def test_list():
...
+    ]
+
+""".encode('utf-8')
+    assert_command(['dcos', 'help', 'package'],
+                   stdout=stdout)
+
+
+def test_help_service():
+    stdout = """Manage DCOS services
+
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: this too please

File: cli/tests/integrations/test_help.py [hunk 5]
```diff
@@ -49,3 +50,263 @@ def test_list():
...
+    <service-id>  The ID for the DCOS Service
+
+""".encode('utf-8')
+    assert_command(['dcos', 'help', 'service'],
+                   stdout=stdout)
+
+
+def test_help_task():
+    stdout = """Manage DCOS tasks
+
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: same

File: cli/dcoscli/help/main.py [hunk 6]
```diff
@@ -45,33 +46,72 @@ def _cmds():
...
+    """
+    :param command: the command name for which you want to see a help
+    :type command: str
+    :returns: process return code
+    :rtype: int
+    """
+
+    try:
+        module = "dcoscli.{}.main".format(command)
+        emitter.publish(__import__(module, fromlist=['']).public_help())
```
Comment (jsancio) [inline cli/dcoscli/help/main.py]: Can we instead get the help command by running `dcos <command> --help` and capturing the stdout? I think this mechanism won't work for command that get install by `dcos package install --cli ...`. Thanks!

**Comment by tamarrow** [comment]
@truskovskiyk looks good. just a little refactoring to keep the tests files easier to read.

**Comment by truskovskiyk** [comment]
@tamarrow  and @jsancio  PR updated, please review

**Comment by truskovskiyk** [comment]
@tamarrow  and @jsancio  PR updated, please review

File: cli/dcoscli/common.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        stdin=stdin,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        env=env)
+
+    # This is needed to get rid of '\r' from Windows's lines endings.
+    stdout, stderr = [std_stream.replace(b'\r', b'')
+                      for std_stream in process.communicate()]
+
+    return (process.returncode, stdout, stderr)
```
Comment (tamarrow) [inline cli/dcoscli/common.py]: This is a duplicate of `exec_command` in cli/tests/integrations/common.py. Can we reuse the code please?

File: cli/dcoscli/help/main.py [hunk 8]
```diff
@@ -46,33 +48,71 @@ def _cmds():
...
+    """
+    :param command: the command name for which you want to see a help
+    :type command: str
+    :returns: process return code
+    :rtype: int
+    """
+
+    returncode, stdout, stderr = exec_command(['dcos', command, '--help'])
+    if returncode != 0:
+        emitter.publish(stderr.decode("utf-8"))
```
Comment (tamarrow) [inline cli/dcoscli/help/main.py]: why do we need to decode here?

File: cli/tests/integrations/test_help.py [hunk 9]
```diff
@@ -49,3 +50,39 @@ def test_list():
...
+def test_help_service():
+    with open('tests/data/help/service.txt') as content:
+        assert_command(['dcos', 'help', 'service'],
+                       stdout=content.read().encode('utf-8'))
+
+
+def test_help_task():
+    with open('tests/data/help/task.txt') as content:
+        assert_command(['dcos', 'help', 'task'],
+                       stdout=content.read().encode('utf-8'))
```
Comment (tamarrow) [inline cli/tests/integrations/test_help.py]: Can we also test a negative case please? dcos help bogus?

**Comment by tamarrow** [comment]
@truskovskiyk looks good, just a few comments.

**Comment by tamarrow** [comment]
@truskovskiyk bump

File: cli/dcoscli/common.py (hunk 7 above)
Comment (truskovskiyk) [inline cli/dcoscli/common.py]: I don't think that import from cli/tests/integrations/common.py is good idea, so I can define exec_command in cli/dcoscli/common.py and then import it to test, is it good?

File: cli/dcoscli/help/main.py (hunk 8 above)
Comment (truskovskiyk) [inline cli/dcoscli/help/main.py]: @tamarrow We need decode because without it, we will have problem with compatible python2/3
